plugins/GrapesJsBuilderBundle/EventSubscriber/EmailSubscriber.php

```diff
@@ -54,4 +81,35 @@ public function onEmailDelete(Events\EmailEvent $event): void
             $this->grapesJsBuilderModel->getRepository()->deleteEntity($grapesJsBuilder);
         }
     }
+
+    public function manageEmailDraft(Events\EmailEditSubmitEvent $event): void
+    {
+        if (!$this->config->isPublished()) {
+            return;
+        }
+
+        $email           = $event->getCurrentEmail();
+        $grapesJsBuilder = $this->grapesJsBuilderModel->getRepository()->findOneBy(['email' => $email]);
+
+        if ($event->isSaveAsDraft()) {
+            // Set draft MJML and restore previous version when saving a draft
+            $grapesJsBuilder->setDraftCustomMjml($grapesJsBuilder->getCustomMjml());
+            $grapesJsBuilder->setCustomMjml($this->existingMjml);
+
+            // reset the html of the parent email as well
+            $email->setCustomHtml($this->existingHtml);
+        }
+
+        if ($event->isApplyDraft()) {
+            // Remove the draft version when applying - the customMjml is already up to date
+            $grapesJsBuilder->setDraftCustomMjml(null);
```

Comment:
When applying the draft, the value of `DraftCustomMjml` should be transferred to `CustomMjml`, and then `DraftCustomMjml` should be set to null. However, I couldn't find any code that updates `CustomMjml`.

Reply:
The customMjml is already up to date when applying the draft. I originally had it as you stated but it is not needed. Please test to verify.